File: spring-bootstrap-java-audit-swagger/src/main/java/com/marcskow/spring/bootstrap/java/pet/Pet.java
package com.marcskow.spring.bootstrap.java.pet;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.marcskow.spring.bootstrap.java.audit.Audit;
import com.marcskow.spring.bootstrap.java.person.model.Person;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
public class Pet extends Audit {

    @Id
    @Column(name = "pet_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String name;

    private String type;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "person_id", nullable = false)
    @JsonIgnore
    private Person person;
}

File: spring-bootstrap-complex-criteria-api/src/main/kotlin/com/marcskow/complex/criteria/api/pet/Pet.kt
package com.marcskow.complex.criteria.api.pet

import com.marcskow.complex.criteria.api.person.model.Person
import javax.persistence.*

@Entity
@Table(name = "pets")
data class Pet(
        @Id
        @Column(name = "pet_id")
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Long = 0,

        var name: String? = null,

        var type: String? = null,

        @ManyToMany(mappedBy = "pets")
        var persons: List<Person> = emptyList(),

        @ManyToMany
        @JoinTable(
                name = "pets_attributes",
                joinColumns = [JoinColumn(name = "pet_id", referencedColumnName = "pet_id")],
                inverseJoinColumns = [JoinColumn(name = "attribute_id", referencedColumnName = "attribute_id")])
        var attributes: List<PetAttribute> = emptyList()
)